src/OrbitBase/GetProcessIdsLinux.cpp

```diff
@@ -86,4 +88,30 @@ std::vector<pid_t> GetTidsOfProcess(pid_t pid) {
   return tids;
 }
 
-}  // namespace orbit_base
\ No newline at end of file
+ErrorMessageOr<pid_t> GetTracerPidOfProcess(pid_t pid) {
+  std::string status_file_name = absl::StrFormat("/proc/%i/status", pid);
+  OUTCOME_TRY(status_file_content, orbit_base::ReadFileToString(status_file_name));
+  std::istringstream status_file_content_ss{status_file_content};
+  std::string line;
+  std::optional<pid_t> tracer_pid;
+  constexpr const char* kTracerPidStr = "TracerPid:";
+
+  while (std::getline(status_file_content_ss, line)) {
+    if (line.find(kTracerPidStr) == std::string::npos) continue;
+    int potential_pid;
+    if (!absl::SimpleAtoi(line.substr(line.find_last_of(kTracerPidStr) + 1), &potential_pid)) {
+      return ErrorMessage(absl::StrFormat("Could not extract pid from line %s", line));
+    }
+    tracer_pid = potential_pid;
```

Comment:
You could just `return potential_pid;` here. That would get rid of `tracer_pid` entirely and after the loop there would just be an `return ErrorMessage(...`.

It's a bit shorter but equivalent to what is already there - so feel free to ignore.